conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/TypeMarker.java

```diff
@@ -42,11 +42,24 @@ protected TypeMarker() {
                 genericSuperclass instanceof ParameterizedType,
                 "Class is not parameterized",
                 SafeArg.of("class", genericSuperclass));
-        type = ((ParameterizedType) genericSuperclass).getActualTypeArguments()[0];
+        Type typeArgument = ((ParameterizedType) genericSuperclass).getActualTypeArguments()[0];
+        Preconditions.checkArgument(
+                !(typeArgument instanceof TypeVariable),
+                "TypeMarker does not support variable types",
+                SafeArg.of("typeVariable", typeArgument));
+        this.type = typeArgument;
+    }
+
+    private TypeMarker(Type type) {
         Preconditions.checkArgument(
                 !(type instanceof TypeVariable),
                 "TypeMarker does not support variable types",
                 SafeArg.of("typeVariable", type));
+        this.type = type;
+    }
+
+    public static <T> TypeMarker<T> of(Type type) {
+        return new TypeMarker<>(type) {};
```

Comment:
Is there something we can do to make this more type safe? I don't think so since we're operating with runtime time values at this point.